akvo/rsr/tests/views/py_reports/test_send_report_via_email.py

```diff
@@ -48,3 +48,31 @@ def test_send_report_via_email(self, url_name, query_params, report_name):
         self.assertEqual(0, EmailReportJob.objects.filter(finished_at__isnull=True).count())
         msg = mail.outbox[0]
         self.assertEqual([self.user.email], msg.to)
+
+    def test_send_report_via_djangoq_email(self):
+        from akvo.rsr.views.py_reports.program_overview_excel_report import handle_email_report
+        self.c.get(f"{reverse('py-reports-program-overview-table', args=(self.program.id,))}")
+
+        # We don't use the old job, so it shouldn't have been scheduled
+        job = EmailReportJob.objects.first()
+        self.assertIsNone(job)
+
+        # Check that the task was enqueued with django-q
+        enqueued_task = OrmQ.objects.first()
+        self.assertIsNotNone(enqueued_task)
+        task_dict = SignedPackage.loads(enqueued_task.payload)
+        self.assertEquals(task_dict.get("name"), "program_overview_excel_report")
+
+        # And with the correct program
+        task_args = task_dict.get("args")
+        params_arg = next(iter(task_args), {})
+        self.assertEquals(self.program.id, params_arg.get("program_id"),
+                          msg="The expected program ID isn't present in the task's first argument")
+
+        # Emulate executing the task without going through django-q
+        # There's currently no easy way to do so
+        handle_email_report(*task_args)
```

Comment:
I think it's better to call the function saved on the task object to make it easier to parameterize later.

```suggestion
        f = task_dict.get("func") 
        f(*task_dict.get("args"), **task_dict.get("kwargs"))
```